src/components/book-form.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import * as actions from '../actions/index';
+
+const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+let identifier = 0;
+
+class BookForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleFormSubmit = this.handleFormSubmit.bind(this);
+  }
+
+  handleFormSubmit(e) {
+    e.preventDefault();
+    const { handleAdding } = this.props;
+    const book = {
+      id: identifier,
+      title: this.titleInput.value,
+      category: categories[this.categoryValue.options.selectedIndex],
+    };
+    identifier += 1;
```

Comment:
wonderful attempt 👏  but this is not how it should be done. you should be using state and bind the input to it. update it with and onChange event and on submit call a dispatch method with the state. **But more on that in the next milestone.**